From 1670cac2b1e8d9b4bb2a0d3910262a226d1c4f64 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 08 Aug 2021 11:20:02 +0300
Subject: [PATCH] [SW-0] gaudi: security: close cfg high gap with rr

Add RR and close the CFG HIGH area 0x7FFD000000-0x7FFE000000.
This way we are safe in that gap region.

Change-Id: Ica8005e5d190f07b3dda14b7dfd8439ff5769d7d
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi/arm/security.c b/arch/gaudi/arm/security.c
index c488a7c..887cb52 100644
--- a/arch/gaudi/arm/security.c
+++ b/arch/gaudi/arm/security.c
@@ -1472,8 +1472,12 @@
 	LBW_RR_NIC1_TS_NIC2_MAC_CH,
 	LBW_RR_NIC3_MAC_CH,
 	LBW_RR_NIC3_TS_NIC4_MAC_CH,
+	LBW_RR_CFG_HIGH_GAP,
 	LBW_RR_DBG_ETR,
 };
+
+#define CFG_HIGH_GAP_BASE	0x7FFD000000ull
+#define DBG_BASE		mmMME_S_ROM_TABLE_BASE
 
 static void preboot_secure_lbw(void)
 {
@@ -1544,6 +1548,11 @@
 	max = mmNIC4_MAC_CORE_BASE;
 	preboot_secure_lbw_rr(LBW_RR_NIC3_TS_NIC4_MAC_CH, min, max);
 
+	/* CFG HIGH GAP: 0x7FFD000000 - 0x7FFE000000 */
+	min = CFG_HIGH_GAP_BASE - 1;
+	max = DBG_BASE;
+	preboot_secure_lbw_rr(LBW_RR_CFG_HIGH_GAP, min, max);
+
 	/* mmPSOC_ETR_AXICTL (single register): RR(LBW_RR_ETR) */
 	min = HL_CFG_BASE + mmPSOC_ETR_AXICTL - 1;
 	max = HL_CFG_BASE + mmPSOC_ETR_AXICTL + sizeof(u32);
